### ISSUE TITLE
Cannot complete registration or any other process requiring email and validations.

### ISSUE THREAD

**Issue by Unsigned-Char** [Aguafrommars/TheIdServer#494]
**Describe the bug**
1. The documentation is not clear about using SendGrid of another email provider.  Providing working elements for each case would have a significant impact since each case seems to be partially implemented.

3. After implementing an email client, the registration and other processes like email updates and password changes, etc., that rely on security tokens to be generated, wouldn't and cannot work under the current architecture.  

-     Mainly, the UserManager and TokenProvider are different instances between the generation and validation of those tokens since they are disposed of and instantiated on each use of services.

-     The tokens must be validated by the same instance that generated them and the user must also have an unchanged SecurityStamp otherwise the TokenProvider validation process always returns InvalidToken.

**To Reproduce**
1. Run TheIdServer
2. Register a new user, rinse and repeat.

**Expected behavior**
A functional registration and validation process where the user receives an email to confirm its address or changes.
The link he receives and the token it contains must be valid upon confirmation.

**Logs**
If applicable, add logs to help explain your problem.

**Platform (please complete the following information):**
 - OS: Windows 10 Pro
 - .Net Core Version 5.0.9

**Additional context**

![image](https://user-images.githubusercontent.com/38705069/129302364-150be43a-9b87-4769-9dd6-fb4113fca53a.png)

![image](https://user-images.githubusercontent.com/38705069/129303258-ae2f96b6-e2b9-44d8-9036-b84dd2adfb74.png)

![image](https://user-images.githubusercontent.com/38705069/129303424-3275ebe6-e94e-4b24-a4d2-63d3e6e9cf92.png)

![image](https://user-images.githubusercontent.com/38705069/129303536-34cd8c37-98be-42a4-9360-d867dcfc97c9.png)

**Comment by aguacongas** [comment]
Hi @Unsigned-Char , thx to contribute again.  
Can you post your configuration and tell me how you implement the service please ?

**Comment by Unsigned-Char** [comment]
The following : 

- Windows 10 Pro
- Clean Visual Studio 2019
- dot.NET SDK 5.0.9
- A clone of Master
- No change to appSettings.json except adding my SendGrid credentials.

I simply compiled the solution and ran TheIdServer in debug mode.  

I used the admin interface to register and change the email address afterward.  In either case, the token validation always returns an invalid token response.

ConfirmEmailChange.cshtml.cs : 48:13 : **var result = await _userManager.ChangeEmailAsync(user, email, code);**

ConfirmEmail.cshtml.cs : 43:13 : **var result = await _userManager.ConfirmEmailAsync(user, code);**

I don't have the debug information for Microsoft's UserManager, but I'm guessing the tokens cannot be validated because they cannot be found, the validators or stores are different instances than those when they were created. 

![image](https://user-images.githubusercontent.com/38705069/129930758-c6393f47-5347-48fb-bb5c-0ca4c8098fe0.png)
